### ISSUE TITLE
`sign_planetary_computer` can reuse tokens across items

### ISSUE THREAD

**Issue by TomAugspurger** [brazil-data-cube/rstac#132]
We've been seeing a large number of rate limiting errors, some of which are coming from sits / rstac users making a large number of requests to the SAS endpoint via `sign_planetary_computer`.

If I understand the current implementation correctly, [exists_token](https://github.com/brazil-data-cube/rstac/blob/bc05e76b390c1971d53ce7c59dda1c28135d5b81/R/signatures.R#L197-L199) will reuse tokens when signing all the things in an object (assets in an item, or items in a feature collection). However, it won't necessarily reuse tokens when calling `sign_planetary_computer` multiple times.

```R
library(rstac)
stac_obj <- stac("https://planetarycomputer.microsoft.com/api/stac/v1/") %>%
   stac_search(collections = "sentinel-2-l2a",
               bbox = c(-47.02148, -17.35063, -42.53906, -12.98314)) %>%
   get_request()

# signing each asset href
x <- stac_obj %>% items_sign(sign_fn = sign_planetary_computer())
y <- stac_obj %>% items_sign(sign_fn = sign_planetary_computer())
```

If we print out the SAS tokens for those (trimmed to just show the signature for brevity), we'll see that 

1. All the assets in `x`'s first item were signed with the same token
2. All the items in `x` were signed with the same token
3. The items in `y` were signed with a new token

```
> print(x$features[[1]]$assets$B01$href) [1] "https://sentinel2l2a01.blob.core.windows.net/...&sig=ZlGV0gZnjI7zTqOhmTe9vfGvh%2B%2B31Ba1cDzVA61WNGI%3D"
> print(x$features[[1]]$assets$B02$href) [1] "https://sentinel2l2a01.blob.core.windows.net/...&sig=ZlGV0gZnjI7zTqOhmTe9vfGvh%2B%2B31Ba1cDzVA61WNGI%3D"
> print(x$features[[2]]$assets$B01$href [1] "https://sentinel2l2a01.blob.core.windows.net/...&sig=ZlGV0gZnjI7zTqOhmTe9vfGvh%2B%2B31Ba1cDzVA61WNGI%3D")
> print(y$features[[1]]$assets$B01$href) [1] "https://sentinel2l2a01.blob.core.windows.net/...&sig=bj3wkpeYRhX0nKvojFLde7iP6z3tvY7SCHwknLTATIY%3D"
```

So it seems like a user will get a new SAS token per call to `sign_planetary_computer()`. Ideally they'd only get one token (or one token until it's close to expiring, at which point it would renew)

**Comment by rolfsimoes** [comment]
Thank you for bringing the rstac package token renewal issue to our attention. We have identified and fixed the problem (#133).

Our solution ensures that SAS tokens are reused as much as possible to minimize the number of requests made. We have made a hotfix to the rstac package that incorporates these changes.
To install the hotfix version, run:
```r
devtools::install_github("brazil-data-cube/rstac")
```

We'll submit to CRAN as soon as possible. We would greatly appreciate it if you could test the new version and let us know if you find any problems.

**Comment by TomAugspurger** [comment]
Great, thanks!
